FastSimulation/SimplifiedGeometryPropagator/plugins/FastTrackDeDxProducer.cc

```diff
@@ -0,0 +1,218 @@
+//    Code Updates:  loic Quertenmont (querten)
+//         Created:  Thu May 10 14:09:02 CEST 2008
+//
+//
+
+
+// system include files
+//#include "RecoTracker/DeDx/plugins/DeDxEstimatorProducer.h"
+#include "FastSimulation/SimplifiedGeometryPropagator/interface/FastTrackDeDxProducer.h"
+
+using namespace reco;
+using namespace std;
+using namespace edm;
+
+bool nothick = false;
+
+
+//void yuval(const std::string& a);
+
+void FastTrackDeDxProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<string>("estimator","generic");
+  desc.add<edm::InputTag>("tracks",edm::InputTag("generalTracks"));
+  desc.add<bool>("UsePixel",false); 
+  desc.add<bool>("UseStrip",true); 
+  desc.add<double>("MeVperADCPixel",3.61e-06*265);
+  desc.add<double>("MeVperADCStrip",3.61e-06);
+  desc.add<bool>("ShapeTest",true);      
+  desc.add<bool>("UseCalibration",false);  
+  desc.add<string>("calibrationPath", "");
+  desc.add<string>("Reccord", "SiStripDeDxMip_3D_Rcd");
+  desc.add<string>("ProbabilityMode", "Accumulation");
+  desc.add<double>("fraction", 0.4);
+  desc.add<double>("exponent",-2.0);
+  desc.add<bool>("convertFromGeV2MeV",true);
+
+  desc.add<edm::InputTag>("simHits");
+  desc.add<edm::InputTag>("simHit2RecHitMap");
+ 
+
+  descriptions.add("FastTrackDeDxProducer",desc);
+  //yuval("hello");
+}
+
+
+FastTrackDeDxProducer::FastTrackDeDxProducer(const edm::ParameterSet& iConfig)
+: simHitsToken(consumes<edm::PSimHitContainer>(iConfig.getParameter<edm::InputTag>("simHits")))
+, simHit2RecHitMapToken(consumes<FastTrackerRecHitRefCollection>(iConfig.getParameter<edm::InputTag>("simHit2RecHitMap")))
+{
+  produces<ValueMap<DeDxData> >();
+
+  string estimatorName = iConfig.getParameter<string>("estimator");
+  if     (estimatorName == "median")              m_estimator = new MedianDeDxEstimator(iConfig);
+  else if(estimatorName == "generic")             m_estimator = new GenericAverageDeDxEstimator  (iConfig);
+  else if(estimatorName == "truncated")           m_estimator = new TruncatedAverageDeDxEstimator(iConfig);
+  //else if(estimatorName == "unbinnedFit")         m_estimator = new UnbinnedFitDeDxEstimator(iConfig);
+  else if(estimatorName == "productDiscrim")      m_estimator = new ProductDeDxDiscriminator(iConfig);
+  else if(estimatorName == "btagDiscrim")         m_estimator = new BTagLikeDeDxDiscriminator(iConfig);
+  else if(estimatorName == "smirnovDiscrim")      m_estimator = new SmirnovDeDxDiscriminator(iConfig);
+  else if(estimatorName == "asmirnovDiscrim")     m_estimator = new ASmirnovDeDxDiscriminator(iConfig);
+
+  //Commented for now, might be used in the future
+  //   MaxNrStrips         = iConfig.getUntrackedParameter<unsigned>("maxNrStrips"        ,  255);
+
+  m_tracksTag = consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("tracks"));
+
+  //simHitsToken = consumes<edm::PSimHitContainer>(iConfig.getParameter<edm::InputTag>("simHits"));
```

Comment:
remove the commented out lines if not needed, or add a comment which explains why you want to keep in the code

Reply:
removed